code/game/machinery/computer/HolodeckControl.dm

```diff
@@ -22,6 +22,24 @@
 /obj/machinery/computer/HolodeckControl/attack_paw(var/mob/user as mob)
 	return
 
+/obj/machinery/computer/HolodeckControl/attack_ghost(mob/dead/observer/user)
+	if(!linkedholodeck)
+		return
+	var/turf/spawnturf
+	var/list/L = get_area_turfs(linkedholodeck.type)
+	L = shuffle(L)
+	looping_turfs:
+		for(var/turf/T in L)
+			if(!T.density)
```

Comment:
There's a process for this called is_blocked_turf or something like that, so you don't need named loops.